pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/RocksdbMetadataStore.java

```diff
@@ -233,12 +240,15 @@ private RocksdbMetadataStore(String metadataURL, MetadataStoreConfig metadataSto
             throw new MetadataStoreException("Fail to create RocksDB file directory", e);
         }
 
+        readonly = "true".equalsIgnoreCase(
+                System.getProperty(ROCKSDB_READONLY, "false"));
         db = openDB(dataPath.toString(), metadataStoreConfig.getConfigFilePath());
 
         this.optionSync = new WriteOptions().setSync(true);
         this.optionDontSync = new WriteOptions().setSync(false);
         this.optionCache = new ReadOptions().setFillCache(true);
         this.optionDontCache = new ReadOptions().setFillCache(false);
+        dbStateLock = new ReentrantReadWriteLock();
```

Comment:
Any reason to move this here?

Reply:
You would got the NPE when init rocksdb failed,

https://github.com/apache/pulsar/pull/17078/files#diff-d8d2e43e5722098642f3e3b650804e38c12c15cb9f5b92ed2f5a612618498d11R259